### PR TITLE
feat: [pipe-24076]: add some features to comment box

### PR DESCRIPTION
**Pull request by cjlee01**
add some features to comment box
- markdown viewer and write/preview tab
image

https://github.com/user-attachments/assets/5a24796b-dca5-4996-b1c4-e0e63f74f93f

### CODE REVIEW

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx [hunk 1]
```diff
@@ -18,17 +18,14 @@ interface StatusButtonProps {
   onChange: (value: string) => void
 }
 
-const StatusButton: React.FC<StatusButtonProps> = memo(({ codeCommentStatus, onChange }) => (
-  <Select placeholder="Status" onValueChange={onChange}>
-    <SelectTrigger>
-      <SelectValue className="py-2" placeholder={codeCommentStatus} />
-    </SelectTrigger>
+const StatusButton: React.FC<StatusButtonProps> = ({ codeCommentStatus, onChange }) => (
```
Comment (vivek-harness) [inline packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx:21]: Get rid of React.FC?

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx [hunk 2]
```diff
@@ -309,15 +307,17 @@ const PullRequestOverview: React.FC<PullRequestOverviewProps> = ({
...
-                        <PullRequestStatusSelect
-                          refetchActivities={refetchActivities}
-                          commentStatusPullReq={commentStatusPullReq}
-                          comment={{
-                            commentItems: commentItems
-                          }}
-                          pullReqMetadata={pullReqMetadata}
-                          repoId={repoId}
-                        />
+                        <div className="flex gap-x-2" key={`${index}-${payload?.id}`}>
```
Comment (vivek-harness) [inline packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx:310]: Using index in key is not a good idea, may be we can use index only if payload id is null. Something like key={payload?.id || index} ?

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx [hunk 3]
```diff
@@ -263,9 +263,7 @@ const PullRequestOverview: React.FC<PullRequestOverviewProps> = ({
                                 contentClassName="border-transparent"
                                 content={
                                   <div className="flex py-1">
-                                    <Text size={2} color="primary">
-                                      {commentItem.payload?.payload?.text as string}
-                                    </Text>
+                                    <MarkdownViewer source={commentItem.payload?.payload?.text || ''} />
                                   </div>
                                 }
                                 key={`${commentItem.id}-${commentItem.author}-${idx}`}
```
Comment (vivek-harness) [inline packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx:269]: Same here, can avoid index and use only if needed ?

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx [hunk 4]
```diff
@@ -18,17 +18,14 @@ interface StatusButtonProps {
...
     <SelectContent>
       <SelectItem value={CodeCommentState.ACTIVE}>Active</SelectItem>
       <SelectItem value={CodeCommentState.RESOLVED}>Resolved</SelectItem>
     </SelectContent>
   </Select>
-))
+)
 StatusButton.displayName = 'StatusButton'
 
 export const PullRequestStatusSelect: React.FC<CodeCommentStatusSelectProps> = ({
```
Comment (vivek-harness) [inline packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx:31]: See if we can avoid using React.FC for all instances in your changes.

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx (hunk 2 above)
Comment (abhinavrastogi-harness) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx:310]: `payload.id` should be sufficient if it's unique.
if it's empty, we should not be rendering it at all, right?

**Review by abhinavrastogi-harness — APPROVED** [review]

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx (hunk 4 above)
Comment (cjlee01) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx:31]: done

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx (hunk 1 above)
Comment (cjlee01) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-status-select-button.tsx:21]: done

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx (hunk 2 above)
Comment (cjlee01) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx]: done

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx (hunk 3 above)
Comment (cjlee01) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx]: done

File: packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx (hunk 2 above)
Comment (cjlee01) [reply packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-overview.tsx]: correct we shud not be rendering it

**Pull request merged**